### PR TITLE
Add e2e test framework

### LINKED ISSUE #69: Add e2e test for slime module
**Is your feature request related to a problem? Please describe.**
Lack of an effective functional testing program.

**Describe the solution you'd like**
Add e2e test framework and writing specific demo for each slime module. 

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by cywang1905**
Add e2e test framework and specific test case for lazyload module.
resolve #69

### CODE REVIEW

File: test/e2e/lazyload_test.go [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+func updateSidecar(f *framework.Framework) {
+	sidecarGroup := "networking.istio.io"
+	sidecarVersion := "v1beta1"
+	sidecarResource := "sidecars"
+	sidecarName := "productpage"
+
+	pods, err := f.ClientSet.CoreV1().Pods(nsApps).List(context.TODO(), metav1.ListOptions{})
+	framework.ExpectNoError(err)
+	for _, pod := range pods.Items {
+		if strings.Contains(pod.Name, "ratings") {
```
Comment (YonkaFang) [inline test/e2e/lazyload_test.go:258]: curl from every ratings pods?

Comment (YonkaFang) [reply test/e2e/lazyload_test.go:258]: we'd better send a single req and confirm that it's sent to global sidecar at that point.

File: test/e2e/lazyload_test.go [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+				sidecarUpdated = true
+				break Loop
+			}
+		}
+		time.Sleep(500 * time.Millisecond)
+	}
+	if !sidecarUpdated {
+		framework.Failf("sidecar updated failed\n")
+	}
+	ginkgo.By("sidecar updated successfully")
```
Comment (YonkaFang) [inline test/e2e/lazyload_test.go:292]: we also need to confirm that the traffic will not be routed to global sidecar while sidecar cr is created and effective.

**Review by YonkaFang — APPROVED** [review]

File: test/e2e/lazyload_test.go (hunk 2 above)
Comment (cywang1905) [reply test/e2e/lazyload_test.go:292]: Ok. I will consider using istio-proxy access log or related prometheus data change to verify this result.

File: test/e2e/lazyload_test.go (hunk 1 above)
Comment (cywang1905) [reply test/e2e/lazyload_test.go:258]: Ok，I will change this part in the future.

**Pull request merged**